GoogleChromeLabs/ps-analysis-tool
# Feature: Add RWS json generator form

## Description
This PR includes code that adds RWS JSON generator form inside RWS panel enabling the user to create sets for `.well-known/first-party-set.json` and to create a PR. It takes basic data (primary domain, contact email, associated sites, service sites, and ccTLD sites), processes the data, and outputs JSON sets with basic validations.



## Relevant Technical Choices
- Implements custom hook, to handle the form input changes and error handling workflow.


## Testing Instructions
- Open any URL and inspect it.
- Open `Site Boundaries` => `Related Website Sets`, and click on `Generate RWS JSON Resources`.
- A form will open up, and you can tinker with values to check error handling and set creation.


## Additional Information:



## Screenshot/Screencast

https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/58820001/95d3e399-f47c-427d-97b9-57f61ad93500




---



Partially addresses #120
